hbase_storage/src/main/java/com/splicemachine/access/hbase/HBasePartitionAdmin.java

```diff
@@ -595,6 +595,45 @@ public boolean replicationEnabled(String tableName) throws IOException {
         return false;
     }
 
+    @Override
+    public void setCatalogVersion(long conglomerateNumber, String version) throws IOException {
+
+        TableName tn = tableInfoFactory.getTableInfo(Long.toString(conglomerateNumber));
+        try {
+            org.apache.hadoop.hbase.client.TableDescriptor td = admin.getDescriptor(tn);
+            ((TableDescriptorBuilder.ModifyableTableDescriptor) td).setValue(SIConstants.CATALOG_VERSION_ATTR, version);
+            boolean tableEnabled = admin.isTableEnabled(tn);
+            if (tableEnabled) {
+                admin.disableTable(tn);
+            }
+            admin.modifyTable(td);
+        }
+        finally {
+            int wait = 0;
+            if (tn != null && !admin.isTableEnabled(tn)) {
+                admin.enableTable(tn);
+                while (wait < 2000 && !admin.isTableEnabled(tn)) {
+                    try {
+                        Thread.sleep(100);
+                        wait++;
+                    } catch (InterruptedException e) {
+                        throw new IOException(e);
+                    }
+                }
```

Comment:
maybe you should again after exiting the while loop that the table is enabled (and throw exception if it is not), because `wait` could've reached 2000 retries and the loop exited however the table is still not enabled yet.

Reply:
Good point!

Reply:
Good point!